subgraph-api/src/api-server.ts

```diff
@@ -0,0 +1,178 @@
+  if ([hemi.id, hemiSepolia.id].includes(req.data.chainId)) {
+    next()
+  } else {
+    next('route')
+  }
+}
+
+const isInteger = string => /^\d+$/.test(string)
+
+function validateQueryParams(req, res, next) {
+  const { fromBlock, limit, skip } = req.query
+
+  req.data = req.data || {}
+
+  if (!isInteger(fromBlock)) {
+    sendJsonResponse(res, 400, { error: 'fromBlock must be a number' })
+    return
+  }
+  req.data.fromBlock = Number.parseInt(fromBlock)
+
+  if (limit) {
+    if (!isInteger(limit)) {
+      sendJsonResponse(res, 400, { error: 'limit must be a number' })
+      return
+    }
+    req.data.limit = Number.parseInt(limit)
+  }
+
+  if (skip) {
+    if (!isInteger(skip)) {
+      sendJsonResponse(res, 400, { error: 'skip must be a number' })
+      return
+    }
+    req.data.skip = Number.parseInt(skip)
+  }
+
+  next()
+}
+
+export function createApiServer() {
+  const app = express()
+
+  app.use(cors({ origin: config.get('originList') }))
+
+  app.get(
+    '/:chainIdStr(\\d+)/:operation(deposits|withdrawals)/meta',
+    parseChainId,
+    function (req, res, next) {
+      const { operation } = req.params
+      if (operation === 'deposits') {
+        validateChainIsEthereum(req, res, next)
+      } else {
+        validateChainIsHemi(req, res, next)
+      }
+    },
+    function (req, res, next) {
+      // @ts-expect-error: req.data.chainId is set in parseChainId
+      const { chainId } = req.data
+
+      getLastIndexedBlock(chainId)
+        .then(function (number) {
+          sendJsonResponse(res, 200, { number })
+        })
+        // eslint-disable-next-line promise/no-callback-in-promise
+        .catch(next)
+    },
+  )
+
+  app.get(
+    '/:chainIdStr(\\d+)/deposits/:address(0x[0-9a-fA-F]{40})',
+    parseChainId,
+    validateChainIsEthereum,
+    validateQueryParams,
+    function (req, res, next) {
+      const { address } = req.params
+
+      // @ts-expect-error: req.data is populated by parseQueryParams
+      getEvmDeposits({ address, ...req.data })
+        .then(function (deposits) {
+          sendJsonResponse(res, 200, { deposits })
+        })
+        // eslint-disable-next-line promise/no-callback-in-promise
+        .catch(next)
+    },
+  )
+
+  app.get(
+    '/:chainIdStr(\\d+)/withdrawals/:address(0x[0-9a-fA-F]{40})/:type(btc|evm)',
+    parseChainId,
+    validateChainIsHemi,
+    validateQueryParams,
+    function (req, res, next) {
+      const { address, type } = req.params
+
+      let getWithdrawals
+      if (type === 'evm') {
+        getWithdrawals = getEvmWithdrawals
+      } else {
+        getWithdrawals = getBtcWithdrawals
+      }
+
+      // @ts-expect-error: req.data is populated by parseQueryParams
+      getWithdrawals({ address, ...req.data })
+        .then(function (withdrawals) {
+          sendJsonResponse(res, 200, { withdrawals })
+        })
+        // eslint-disable-next-line promise/no-callback-in-promise
+        .catch(next)
```

Comment:
Considering this is `express` and a common pattern when using this library, perhaps we could disable this rule for the entire `subgraph-api`. What do you think?

Reply:
Or return the promise. That may work too.

Reply:
Do not work. This is for `express@5` 🤷🏼‍♂️